compiler/src/dotty/tools/dotc/typer/Applications.scala

```diff
@@ -1816,17 +1836,15 @@ trait Applications extends Compatibility {
             isAsGood(alt1, tp1.instantiate(tparams.map(_.typeRef)), alt2, tp2)
           }
         case _ => // (3)
-          def isGiven(alt: TermRef) =
-            alt1.symbol.is(Given) && alt.symbol != defn.NotGivenClass
-          def compareValues(tp1: Type, tp2: Type)(using Context) =
-            isAsGoodValueType(tp1, tp2, isGiven(alt1), isGiven(alt2))
+          def compareValues(tp2: Type)(using Context) =
+            isAsGoodValueType(tp1, tp2, alt1.symbol.is(Implicit))
```

Comment:
This change comes from https://github.com/scala/scala3/pull/21226 that was not backported to 3.5.0
@odersky Can you confirm that it's expected and should remain in 3.5?

Reply:
Confirmed in direct message